clang-tools-extra/clang-tidy/utils/FormatStringConverter.cpp

```diff
@@ -230,6 +241,50 @@ FormatStringConverter::FormatStringConverter(ASTContext *ContextIn,
   finalizeFormatText();
 }
 
+std::optional<StringRef>
+FormatStringConverter::formatStringContainsUnreplaceableMacro(
```

Comment:
> getting the macro name for the function call and permitting that macro for the format string too

>  such as the same macro being intentionally used in both places

It should be possible to detect this case comparing the `FileID` of the call's location and the format expression's location. If they differ, then they are different arguments -> different macros. Although, I haven't checked that yet.